modules/develop/pages/data-transforms/build.adoc

```diff
@@ -101,9 +108,54 @@ fn my_transform(event: WriteEvent, writer: &mut RecordWriter) -> Result<(), Box<
 }
 ----
 --
-======
+JavaScript::
++
+--
+All transform functions must register a callback with the `onRecordWritten()` method.
+
+You should run any initialization steps outside of the callback so that they are only run once when the transform function is first deployed.
+
+[source,js]
+----
+// src/index.js
+import { onRecordWritten } from "@redpanda-data/transform-sdk";
+
+// This is a good place to perform setup steps.
+
+// Register your transform function.
+onRecordWritten((event, writer) => {
+  //This is where you read the record that was written, and then you can
+  // return new records that will be written to the output topic
+  writer.write(event.record);
+});
+----
+
+If you need to use Node.js standard modules in your transform function, you must configure the https://github.com/cyco130/esbuild-plugin-polyfill-node[`polyfillNode` plugin] for https://esbuild.github.io/[esbuild]. This plugin allows you to polyfill Node.js APIs that are not natively available in the Redpanda JavaScript runtime environment.
 
-When creating a transform function, you should run any initialization steps in the `main()` function because it's only run once when the transform function is first deployed. If you're using Go, you can also use the standard predefined https://go.dev/doc/effective_go#init[`init()` function].
+.`esbuild.js`
+[source,js]
+----
+import * as esbuild from 'esbuild';
+import { polyfillNode } from 'esbuild-plugin-polyfill-node';
+
+await esbuild.build({
+  plugins: [
+    polyfillNode({
+      globals: {
+        buffer: true, // Allow a global Buffer variable if referenced.
+        process: false, // Don't inject the process global, the Redpanda JavaScript runtime does that.
+      },
+      polyfills: {
+        console: true, // Enable console polyfill
```

Comment:
@rockwotj you mentioned that `console` is included. I wonder if we should explicitly add this to `polyfills` so it's transparent what is/isn't included.